parlai/core/metrics.py

```diff
@@ -251,6 +251,144 @@ def value(self) -> float:
         return self._numer / self._denom
 
 
+class ConfusionMatrixMetric(Metric, ABC):
+    """
+    Class that keeps count of the confusion matrix and computes precision for
+    classification.
+    """
+
+    __slots__ = (
+        '_true_positives',
+        '_true_negatives',
+        '_false_positives',
+        '_false_negatives',
+    )
+
+    def __init__(
+        self,
+        true_positives: TScalar = 0,
```

Comment:
We could make the whole thing more compact if we just used TP/FP/TN/FN and define them in the docstring, but if you prefer the more verbose names that's cool with me.

Reply:
I prefer the long names for clarity, but that's just me =/